stdio/bignum.c

```diff
@@ -37,13 +38,28 @@ static void bignum_shiftrInternal(bignum_t *bn, const size_t shift)
 
 static int bignum_resize(bignum_t *bn, const size_t newSize)
 {
-	bignum_data_t *ptr = realloc(bn->data, sizeof(*bn->data) * newSize);
-	if (ptr != NULL) {
+	bignum_data_t *ptr;
+	if (newSize <= BIGNUM_STACK_BUFFER_SIZE) {
 		bn->size = newSize;
-		bn->data = ptr;
-		return 0;
 	}
-	return -1;
+	else if (bn->size <= BIGNUM_STACK_BUFFER_SIZE) {
+		ptr = calloc(newSize, sizeof(*bn->data));
+		if (ptr != NULL) {
+			(void)memcpy(ptr, bn->data, bn->size * sizeof(*bn->data));
+			bn->size = newSize;
+			bn->data = ptr;
+			return 0;
+		}
+	}
+	else {
```

Comment:
Case of `bn->size > newSize` and `bn->size == newSize` aren't handled

Reply:
For me missing cases are:
* `bn->size == newSize`  -> no operation 
* `(newSize < bn->Size) && (bn->data != bn->stackBuffer)` -> shrink from `re/c/alloc` to `use-stack`.

I see that way, until `bn->size` and `newSize` are grater than `BIGNUM_STACK_BUFFER_SIZE`, the check `bn->size > newSize` (shrink) and `bn->size < newSize` (grow) are handled (catched) by this `else` block with `realloc()`, but `bn->size == newSize` case should be handled explicitly `just return 0`, of course `realloc()` with reassignement in this block will handle that but it's waste of cycles. However, the case of freeing `realloc'ed` memory when returning to the `use stack memory`case is definitely not handled: `newSize < bn->Size` when `bn->data != bn->stackBuffer`.